### PR TITLE
Add listing of hdf5 file formats to docs

### PR DESCRIPTION
**Pull request by ashander**
as discussed in 325

(once this looks good we can remove the .txt files in the rootdir and squash to one commit)

### CODE REVIEW

File: docs/file-format.rst [hunk 1]
```diff
@@ -535,3 +535,36 @@ Path                               Type
...
+upgrade` will convert older tree sequence files to the latest version.
+
+However many changes to the tree sequence format are not part of major
+releases. The table below gives these versions (contained in the root group
+attribute, ``format_version`` as a pair ``(major, minor)``).
+
+================    =================
+Version             Commit Short Hash
+================    =================
+HDF5_FORMAT_V9.0    e504abd
```
Comment (jeromekelleher) [inline docs/file-format.rst]: We can probably just cut this down to 9.0 etc. here. This is probably more readable I think.

**Comment by jeromekelleher** [comment]
This is great, thanks @ashander. Unfortunately, I don't think we've got it quite right. For example,
```
$ git checkout 52d64ce  # Version 4.0
$ python msp_dev.py simulate 10 tmp.hdf5
$ h5dump tmp.hdf5 | head
HDF5 "tmp.hdf5" {
GROUP "/" {
   ATTRIBUTE "format_version" {
      DATATYPE  H5T_STD_U32LE
      DATASPACE  SIMPLE { ( 2 ) / ( 2 ) }
      DATA {
      (0): 5, 0
      }
   }
```
It seems that what has happened here is that we've got the merge commit itself here (which changed from 4.0 to 5.0) and not its parent (which was the commit it was merged into, i.e., the last version supporting 4.0).

**Comment by ashander** [comment]
doh. thanks for catching that. I've fixed and works in this spot check

```
$ git checkout 62659fb #5.0
$  make clean
$  make
$  python msp_dev.py simulate 10 tmp.hdf5
$ h5dump tmp.hdf5 |head
HDF5 "tmp.hdf5" {
GROUP "/" {
   ATTRIBUTE "format_version" {
      DATATYPE  H5T_STD_U32LE
      DATASPACE  SIMPLE { ( 2 ) / ( 2 ) }
      DATA {
      (0): 5, 0
      }
   }
   ATTRIBUTE "sample_size" {
```

File: docs/file-format.rst [hunk 2]
```diff
@@ -535,3 +535,36 @@ Path                               Type
...
+6.0        6310725
+5.0        62659fb
+4.0        a586646
+3.2        8f44bed
+3.1        d69c059
+3.0        7befdcf
+2.1        a26a227
+2.0        7c507f3
+1.1        04722d8
+1.0        04722d8
```
Comment (ashander) [inline docs/file-format.rst]: these failed because they're the same o_O

File: map_check.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+make rm -fR build && rm -f *.o *.so && make
+python msp_dev.py simulate 10 tmp.hdf5 && h5dump tmp.hdf5 | head >> check_output.txt
+echo "checking 0.3" >> check_output.txt
+git checkout f42215e
+make rm -fR build && rm -f *.o *.so && make
+python msp_dev.py simulate 10 tmp.hdf5 && h5dump tmp.hdf5 | head >> check_output.txt
+echo "checking 0.2" >> check_output.txt
+git checkout 26e049d
+make rm -fR build && rm -f *.o *.so && make
+python msp_dev.py simulate 10 tmp.hdf5 && h5dump tmp.hdf5 | head >> check_output.txt
```
Comment (ashander) [inline map_check.sh]: 2.0 which failed due to

```
Traceback (most recent call last):
  File "msp_dev.py", line 7, in <module>
    msprime.cli.msp_main()
  File "/home/jaime/lib/msprime/msprime/cli.py", line 846, in msp_main
    args.runner(args)
  File "/home/jaime/lib/msprime/msprime/cli.py", line 764, in run_simulate
    random_seed=args.random_seed)
  File "/home/jaime/lib/msprime/msprime/trees.py", line 498, in simulate
    tree_sequence = sim.get_tree_sequence()
  File "/home/jaime/lib/msprime/msprime/trees.py", line 797, in get_tree_sequence
    ll_tree_sequence.create(self._ll_sim)
TypeError: function takes exactly 2 arguments (1 given)
```

File: docs/file-format.rst (hunk 2 above)
Comment (ashander) [reply docs/file-format.rst]: this happened because the changes from 04722d8 to 2.0 were not done on separate branches and merged in

**Comment by jeromekelleher** [comment]
Superb, thanks @ashander! Let's just forget about 2.0, (i.e., delete it from the table). No one will ever use it, so there's no point in spending time on it.

Can you squash the commits please so I can merge?

**Comment by ashander** [comment]
Done. I added a brief comment in the rst too to help any future git
explorers

**Pull request merged**